### ISSUE TITLE
Linux ran out of storage!! :(

### ISSUE THREAD

**Issue by lilyzding** [brittAnderson/Intro2Computing4Psychology#85]
I've been using Ubuntu with hyper-v and recently I've been unable to install anything as I have no storage left (attached file with dev/sda1 being full). I can't figure out how to increase the allotted storage with hyper-v and I don't have extraneous things installed either. Has anyone bumped into this issue before and know how to work around it?

![image](https://user-images.githubusercontent.com/67672282/182988706-cc4bdacf-1a10-4991-b981-7a4cac026158.png)

**Comment by brittAnderson** [comment]
This is a nasty problem. And while it is in theory possible to expand your partition it almost always ends in failure. Two choices. 1) If you have lots of room on your computer hard drive you may find it faster to just make a new virtual machine with more memory. If not you can try removing everything on your ubuntu side that you don't absolutely need. you can do a "sudo apt autoremove". You can also look at how many kernels you have. When your machine starts the ubuntu side does it list lots of versions you could boot into? If so, we can talk about how to get rid of those. 2) Try to increase VM memory. Here is a link with some instructions. But there is a risk to damage the VM so make sure any critical files you want to submit for homeworks have been backed up to a usb or something before trying. Go slow. And go carefully. http://linguist.is/2020/08/12/expand-ubuntu-disk-after-hyper-v-quick-create/

**Comment by K14R45H** [comment]
Aw man. I'm sorry you ran into this issue, Lily. I faced the same problem earlier in the course, messed up and had to reinstall my Ubuntu VM. 

I think Professor Anderson's suggestions are great. Ideally, if you have like 30 - 50 GB of physical storage space to spare on your computer, it may be easier to make a new VM and allocate more storage to it from the start. 

If you have decided to try and expand the virtual storage space allocated to your VM, I would reiterate Professor Anderson's point: go slowly and carefully. Make sure you carry out all the steps and check that they have been done (as the article provided by our professor above outlines). In particular, make sure you successfully expand the virtual storage space in Ubuntu that you allocated in Hyper-V settings.

Good luck and feel free to get in touch again!

**Comment by lilyzding** [comment]
So a little update for @brittAnderson ! I didn't do anything too special here. I moved my psychopy setup out of Linux, and onto a windows setup in the psychopy GUI coder, and uninstalled the wheel I installed with it. Apparently this is almost the exact size of the Latex installation, so I was able to install Latex to finish the rest of the assignments. I also ran a lot of autocleans, removed quite a bit of cache, and sudo rebooted a whole lot. Assignment files didn't need to be removed since the size of each were only a few hundred bytes. 

I hope this can maybe provide at least the suggestion of moving a few components out of Linux to Windows for your next classes!
